sematic/resolvers/state_machine_resolver.py

```diff
@@ -97,12 +113,17 @@ def _set_future_state(self, future, state):
 
     # State machine lifecycle hooks
 
-    def _resolution_will_start(self) -> None:
+    def _resolution_will_start(self, root_future: AbstractFuture) -> None:
```

Comment:
Is `root_future` still necessary here since we have `self._root_future`?

Reply:
oops! Missed this one when I reverted these changes to the event hooks. Updated now